journals/settings/utils.py

```diff
@@ -10,3 +16,18 @@ def get_env_setting(setting):
     except KeyError:
         error_msg = "Set the [%s] env variable!" % setting
         raise ImproperlyConfigured(error_msg)
+
+
+def get_whitelist_domains(request):
+    """
+        Extracts and returns domain list from siteconfiguration urls, including the domain on which server is running.
+    """
+    permitted_domains = ['', '{}:{}'.format(request.site.hostname, request.site.port)]
```

Comment:
Can we use something like this to make it consistent on dev and prod like environments?
```suggestion
    permitted_domains = [urlsplit(request.build_absolute_uri())]
```

Reply:
build_absolute_uri() and get_host() returns unexpected domain for tests i.e "http://testserver/".
but in test I am building expected base_url for request using `self.site` like this
`base_url = "https://{}:{}/".format(self.site.hostname, self.site.port)` 
and results something like https://localhost:18606/
so test fails.

If still we want to use build_absolute_uri , then I think we need to hardcode `base_url` to "http://testserver/".
So what you say ?